### ISSUE TITLE
notFound & error handlers with 2.4.0 class controllers

### ISSUE THREAD

**Issue by calmdev** [slimphp/Slim#741]
Should the notFound and error handlers work with the new [2.4.0](https://github.com/codeguy/Slim/releases/tag/2.4.0) style controllers?

My controller(s):

``` php
// Load controllers.
$home = new \App\Controller\Home($app);
```

This works:

``` php
// Define routes.
$app->get('/', array($home, 'index'));
$app->notFound(array($home, 'notFound'));
$app->error(array($home, 'error'));
```

This works:

``` php
// Define routes.
$app->get('/', '\App\Controller\Home:index');
$app->notFound(array($home, 'notFound'));
$app->error(array($home, 'error'));
```

This does not work:

``` php
// Define routes.
$app->get('/', '\App\Controller\Home:index');
$app->notFound('\App\Controller\Home:notFound');
$app->error('\App\Controller\Home:error');
```

Here's the error I encounter in the last example:

![screen shot 2014-02-01 at 3 41 33 pm](https://f.cloud.github.com/assets/461249/2058734/7d26ce0a-8b81-11e3-8fa6-d8b9e7ef9af0.png)

**Comment by erizocosmico** [comment]
Nope, `notFound` and `error` currently don't support the new style controllers.

**Comment by calmdev** [comment]
Ok, I was wondering if it was intentionally left out for some reason or just overlooked.

I'd like to use your PR #743, so hope it's merged :+1:

**Comment by codeguy** [comment]
Closing. Addressed in PR #752.
